src/Containers/Microsoft.NET.Build.Containers/Tasks/CreateNewImage.cs

```diff
@@ -325,4 +343,109 @@ public void Dispose()
             logWarning: s => Log.LogWarningWithCodeFromResources(s),
             logError: (s, a) => { if (a is null) Log.LogErrorWithCodeFromResources(s); else Log.LogErrorWithCodeFromResources(s, a); });
     }
+
+    private Telemetry CreateTelemetryContext(SourceImageReference source, DestinationImageReference destination)
+    {
+        var context = new PublishTelemetryContext(
+            source.Registry is not null ? GetRegistryType(source.Registry) : null,
+            null, // we don't support local pull yet, but we may in the future
+            destination.RemoteRegistry is not null ? GetRegistryType(destination.RemoteRegistry) : null,
+            destination.LocalRegistry is not null ? GetLocalStorageType(destination.LocalRegistry) : null);
+        return new Telemetry(Log, context);
```

Comment:
I'm not very familiar with this pattern with `Log`. Where is the `Log` object coming from? It appears to be an MSBuildLogger, which the code for telemetry on that thing checks out. But this `Log` appears to be an instance of a `TaskLoggingHelper`. I'm not sure how the two are connected.

Reply:
`Log` in the context of an MSBuild Task is the TaskLoggingHelper, which is a wrapper around the raw MSBuild ILogger interface (Note: this is not the same as it Microsoft.Extension.Logging.ILogger). Main differences are that the TaskLoggingHelper is thread-safe and that it has a few convenience methods. All of this is handled/provided by the MSBuild Engine, nothing we have to do/care about.